src/shared_realm.cpp

```diff
@@ -554,5 +555,110 @@ util::Optional<int> Realm::file_format_upgraded_from_version() const
     return util::none;
 }
 
+#define VERSION_CAST(version_id) {(version_id).version, (version_id).index}
```

Comment:
As far as I can tell, a template function that does the equivalent of this macro is also not useful because the compiler is unable to infer the template arguments.